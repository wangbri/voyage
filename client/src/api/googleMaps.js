import openSocket from 'socket.io-client';
const  socket = openSocket('http://localhost:8000');

var locationsMarkers = []; // keep track of each of the markers from user
// var event = new Event('build');

// locationsMarkers.addEventListener('build', function(e) {

// });

export function sendRoute(input) {
  socket.emit('route', input);
}

export function receiveRoute(cb) {
  socket.on('route', data => cb(null, data));
}

var schedules = [];
export function createSchedule(){
  // for (var i = 0; i < locationsMarkers.length; i++) {
  //   console.log()
  //   //for(var j = 0; j < locationsMarkers.length; j++){
  //     //schdules.push(new Schedule(addresses[i], addresses[j]))
  //   //}
  // }
  for(var startKey in locationsMarkers){
    console.log(startKey, locationsMarkers[startKey]);
    // for(var endKey in locationsMarkers){
    //   schedules.push(new Schedule())
    // }
  }
}

// draw the different directions  
export function calculateAndDisplayRoute(addresses, directionsService, directionsDisplay) {
  var startAddress = addresses[0];
  var endAddress = addresses[addresses.length - 1];
  var waypts = [];

  // use waypoints for intermediate destinations
  for (var i = 1; i < addresses.length - 1; i++) {
    waypts.push({
      location: addresses[i],
      stopover: true
    });
  }

  var directionsParams = {
    origin: startAddress,
    destination: endAddress,
    travelMode: 'DRIVING'
  }

  if (waypts.length > 0) {
    directionsParams['waypoints'] = waypts;
    directionsParams['optimizeWaypoints'] = true;
  }

  directionsService.route(directionsParams, function(response, status) {
    if (status === 'OK') {
      directionsDisplay.setDirections(response);
    } else {
      alert('Directions request failed due to the following reason: ' + status);
    }
  });
}

// uses geoencoding api to generate markers
export function codeAddress(input, geocoder, map) {
  var name;
  var address;
  var image;
  var link;

  if (input != null) {
    name = input['name'];
    address = input['location'];
    image = input['image'];
    link = input['link'];
  }

  geocoder.geocode({'address': address}, function(results, status) {
    if (status == 'OK') {
      map.setCenter(results[0].geometry.location);

      if (input != null) {
        map.setZoom(16);
      }

      var marker = new window.google.maps.Marker({
        map: map,
        position: results[0].geometry.location,
        title: results[0].formatted_address
      });

      locationsMarkers.push({
        key: name,
        value: marker
      });


      if (name == null) {
        name = results[0].formatted_address;
      }

      var infowindow = new window.google.maps.InfoWindow({
        content: '<div class="card" style="width: 18rem;"><img class="card-img-top" style="width: 18rem;" src=' + image + 
        '><div class="card-body"><h5 class="card-title">' + name + 
        '</h5><p class="card-text">Some quick example text to build on the card title and make up the bulk of the card content.</p><a href=' + link + 
        ' class="btn btn-primary">Yelp it</a>&nbsp;<button class="btn btn-danger" onclick="removeMarker(\'' + marker['position'] + 
        '\')">Remove</button></div></div>',   
        maxWidth: 230
      });

      marker.addListener('click', function() {
        infowindow.open(map, marker);
      });
    } else {
      alert('Geocode was not successful for the following reason: ' + status);
    }

    // markers.push(marker);
  });
}

export { locationsMarkers };