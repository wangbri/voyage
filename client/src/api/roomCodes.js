import openSocket from 'socket.io-client';
const socket = openSocket('http://localhost:8000');

export function saveSchedule(input) {
  socket.emit('saveSchedule', input);
}

export function getCode(cb) {
	socket.on('code', data => cb(null, data));
}

export function getSchedule(input) {
  socket.emit('getSchedule', input);
}

export function receiveSchedule(cb) {
  socket.on('getSchedule', data => cb(null, data));
}