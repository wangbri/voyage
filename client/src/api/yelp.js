import openSocket from 'socket.io-client';
const  socket = openSocket('http://localhost:8000');

// export function subscribeToTimer(cb) {
//   socket.on('timer', timestamp => cb(null, timestamp));
//   socket.emit('subscribeToTimer', 1000);
// }

export function sendYelpAddress(input) {
	socket.emit('yelp', input);
}

export function receiveYelpResult(cb) {
	socket.on('yelp', data => cb(null, data));
}