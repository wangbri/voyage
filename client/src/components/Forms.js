import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { sendYelpAddress, receiveYelpResult } from '../api/yelp.js';
import { sendRoute, receiveRoute } from '../api/googleMaps.js';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: "",
      addresses: []
      // map: null,
      // geocoder: null
    }

    this.handleYelpSubmit = this.handleYelpSubmit.bind(this);
    this.handleYelpChange = this.handleYelpChange.bind(this);
    this.handleAddressSubmit = this.handleAddressSubmit.bind(this);
    this.handleAddressChange = this.handleAddressChange.bind(this);

    receiveYelpResult((err, data) => this.props.addMarker(data));
    receiveRoute((err, data) => this.props.displayRoute(data));
  }

  handleYelpChange(event) {
    this.setState({value: event.target.value});
  }

  handleYelpSubmit(event) {
    // window.socket.emit('yelp', this.state.value);
    // console.log(this.state.value);

    sendYelpAddress(this.state.value);
  }

  handleAddressChange(event) {
    var inputName = event.target.id;
    var inputIndex = inputName[inputName.length - 1];
    const addresses = this.state.addresses;
    addresses[inputIndex] = event.target.value;
    this.setState({ addresses: addresses });
    console.log(inputName);
    console.log(this.state.addresses[inputIndex]);
  }

  handleAddressSubmit(event) {
    for (var i = 0; i < 3; i++) {
      console.log(this.state.addresses[i]);
      sendYelpAddress(this.state.addresses[i]);
    }

    sendRoute(this.state.addresses);
  }

  render() {
    return (
      <div id="submitAddress">
        <form className="form-inline">
          <div className="form-group mx-sm-3 mb-2">
            <input className="form-control" id="inputAddress0" placeholder="Specify an address.." value={this.state.addresses[0]} onChange={this.handleAddressChange}></input>
            <input className="form-control" id="inputAddress1" placeholder="Specify an address.." value={this.state.addresses[1]} onChange={this.handleAddressChange}></input>
            <input className="form-control" id="inputAddress2" placeholder="Specify an address.." value={this.state.addresses[2]} onChange={this.handleAddressChange}></input>
          </div>
          <button type="button" className="btn btn-primary mb-2" onClick={this.handleAddressSubmit}>Submit Address</button>

          <div className="form-group mx-sm-3 mb-2">
            <input className="form-control" id="inputYelpAddress" placeholder="Specify a Yelp address.." value={this.state.value} onChange={this.handleYelpChange}></input>
          </div>
          <button type="button" className="btn btn-primary mb-2" id="yelp-btn" onClick={this.handleYelpSubmit}>Submit Yelp Address</button>
        </form>
      </div>
    );
  }
}

export default Form;