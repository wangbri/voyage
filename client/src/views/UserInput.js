import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { sendInputs } from '../api/yelp.js';

class UserInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: "austin, tx",
      price: "$",
      transit: "driving",
      start: "",
      end: ""
    }

    this.sendInputs = this.sendInputs.bind(this);
    this.handleLocationChange = this.handleLocationChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
    this.handleStartChange = this.handleStartChange.bind(this);
    this.handleEndChange = this.handleEndChange.bind(this);
    this.handleTransitChange = this.handleTransitChange.bind(this);
  }

  handleLocationChange(event) {
    this.setState({location: event.target.value});
  }

  handlePriceChange(event) {
    this.setState({price: event.target.value});
  }

  handleStartChange(event) {
    this.setState({start: event.target.value});
  }

  handleEndChange(event) {
    this.setState({end: event.target.value});
  }

  handleTransitChange(event) {
    this.setState({transit: event.target.value});
  }

  sendInputs(event) {
    console.log(this.state.price);
    console.log(this.state.location);
    console.log(this.state.transit);
    console.log(this.state.start);
    console.log(this.state.end);

    sessionStorage.setItem('price', this.state.price);
    sessionStorage.setItem('location', this.state.location);

    sendInputs({
      location: this.state.location,
      price: this.state.price,
      start: this.state.start,
      end: this.state.end,
      transit: this.state.transit
    });
  }

  render() {
    return (
      <div>
        <header className="masthead2 text-white text-center">
          <div className="overlay"></div>
          <div className="container">
            <div className="row">
              <div className="col-md-10 col-lg-8 col-xl-7 mx-auto">
                <div className="input-form">
                  <form className="needs-validation" novalidate>
                    <div className="form-row">
                      <div className="col-md-3 mb-3">
                        <label for="validationCustom01">Where do you plan to travel?</label>
                        <input type="text" className="form-control" id="validationCustom01" placeholder="Specify a location..." value={this.state.location} onChange={this.handleLocationChange} required></input>
                        <div className="valid-feedback">
                          Looks good!
                        </div>
                        <div className="invalid-feedback">
                          Please provide a city or location.
                        </div>
                      </div>
                      <div className="col-md-3 mb-3">
                        <label for="validationCustom02">How much do you want to spend?</label>
                        <select id="inputState" className="form-control" onChange={this.handlePriceChange}>
                          <option defaultValue>$</option>
                          <option>$$</option>
                          <option>$$$</option>
                          <option>$$$$</option>
                        </select>
                        <div className="valid-feedback">
                          Looks good!
                        </div>
                        <div className="invalid-feedback">
                          Please provide a price range.
                        </div>
                      </div>
                      <div className="col-md-3 mb-3">
                        <label for="validationCustomUsername">How will you be traveling?</label>
                          <select id="inputState" className="form-control" onChange={this.handleTransitChange}>
                            <option defaultValue>driving</option>
                            <option>bicycling</option>
                            <option>walking</option>
                          </select>              
                          <div className="invalid-feedback">
                            Please choose a method of transportation.
                          </div>
                      </div>
                    </div>
                    <div className="form-row">
                      <div className="col-md-3 mb-3">
                        <label for="validationCustom03">When will you be starting your trip?</label>
                        <input type="text" className="form-control" placeholder="Specify a starting time..." value={this.state.start} onChange={this.handleStartChange} required></input>
                        <div className="invalid-feedback">
                          Please provide a valid time.
                        </div>
                      </div>
                      <div className="col-md-3 mb-3">
                        <label for="validationCustom04">When will you have to end your trip?</label>
                        <input type="text" className="form-control" placeholder="Specify an ending time..." value={this.state.end} onChange={this.handleEndChange} required></input>
                        <div className="invalid-feedback">
                          Please provide a valid time.
                        </div>
                      </div>
                    </div>
                    <div className="form-row">
                      <div className="col-md-1 mb-1">
                        <button className="btn btn-primary" type="submit" onClick={this.sendInputs}>Send Input</button>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </header>

        <section className="features-icons bg-light text-center">
          <div className="container">
            <div className="row">
              <div className="col-lg-4">
                <div className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3">
                  <div className="features-icons-icon d-flex">
                    <i className="icon-screen-desktop m-auto text-primary"></i>
                  </div>
                  <h3>Fully Responsive</h3>
                  <p className="lead mb-0">This theme will look great on any device, no matter the size!</p>
                </div>
              </div>
              <div className="col-lg-4">
                <div className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3">
                  <div className="features-icons-icon d-flex">
                    <i className="icon-layers m-auto text-primary"></i>
                  </div>
                  <h3>Bootstrap 4 Ready</h3>
                  <p className="lead mb-0">Featuring the latest build of the new Bootstrap 4 framework!</p>
                </div>
              </div>
              <div className="col-lg-4">
                <div className="features-icons-item mx-auto mb-0 mb-lg-3">
                  <div className="features-icons-icon d-flex">
                    <i className="icon-check m-auto text-primary"></i>
                  </div>
                  <h3>Easy to Use</h3>
                  <p className="lead mb-0">Ready to use with your own content, or customize the source files!</p>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}

export default UserInput;