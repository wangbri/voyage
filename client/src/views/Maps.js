import React, { Component } from 'react';
import { codeAddress, calculateAndDisplayRoute } from '../api/googleMaps.js';
import Forms from '../components/Forms';
import ScrollMarkers from '../components/Marker';

class Maps extends Component {
  constructor(props) {
    super(props);

    this.state = {
      map: null,
      geocoder: null,
      directionsService: null,
      directionsDisplay: null,
      markers: []
    }

    this.addMarker = this.addMarker.bind(this);
    this.displayRoute = this.displayRoute.bind(this);
    this.removeMarker = this.removeMarker.bind(this);
    this.setMapOnAll = this.setMapOnAll(this);
  }

  componentDidMount() {
    var map = new window.google.maps.Map(document.getElementById('map'), {
      zoom: 8,
      center: {
        lat: -34.397,
        lng: 150.644
      }
    });

    var geocoder = new window.google.maps.Geocoder();
    var directionsService = new window.google.maps.DirectionsService();
    var directionsDisplay = new window.google.maps.DirectionsRenderer();
    directionsDisplay.setMap(map);

    this.setState({
      map: map,
      geocoder: geocoder,
      directionsService: directionsService,
      directionsDisplay: directionsDisplay
    });
  }

  componentWillMount() {
    // var geocoder = new window.google.maps.Geocoder();
    // var directionsService = new window.google.maps.DirectionsService();
    // var directionsDisplay = new window.google.maps.DirectionsRenderer();

    // this.setState({
    //   geocoder: geocoder,
    //   directionsService: directionsService,
    //   directionsDisplay: directionsDisplay
    // });
  }

  setMapOnAll() {
    var markers = this.state.markers;

    for (var i = 0; i < markers.length; i++) {
      markers[i].marker.setMap(this.state.map);
    }
  }

  addMarker(input) {
    codeAddress(input, this.state.geocoder, this.state.map)
    .then(marker => {
      var markers = this.state.markers;
      // markers.push(input);
      markers.push({
        marker: marker,
        input: input
      });

      this.setState({
        markers: markers
      });

      var infowindow = new window.google.maps.InfoWindow({
        content: '<div class="card" style="width: 18rem;"><img class="card-img-top" src=' + input['image'] + 
        '><div class="card-body"><h5 class="card-title">' + input['name'] + 
        '</h5><p class="card-text">Some quick example text to build on the card title and make up the bulk of the card content.</p><a href=' + input['link'] + 
        ' class="btn btn-primary">Yelp it</a>&nbsp;<button class="btn btn-danger">Remove</button></div></div>',   
        maxWidth: 230
      });

      // WHY does it have to be an arrow function to get scope??
      marker.addListener('click', (e) => {
        infowindow.open(this.state.map, marker);
      });

      console.log("marker input " + input['name']);
      console.log(markers);
      // console.log("addMarker called: " + markers[0].marker.title);
    });

    // console.log(sessionStorage.getItem("myCat"));
    // this.ScrollMarkers.handleMarkerChange();
  }

  removeMarker(position) {
    var markers = this.state.markers;

    for (var i = 0; i < markers.length; i++) {
      if (markers[i].marker.position == position) {
        markers[i].marker.setMap(null);
        markers.splice(i, 1);
      }
    }

    console.log(markers);
    this.setState({ markers: markers });
    // this.setMapOnAll();
  }

  displayRoute(input) {
    calculateAndDisplayRoute(input, this.state.directionsService, this.state.directionsDisplay);
  }

  render() {
    return (
      <div>
        <div id="map" style={{height: 550}}></div>
        <Forms addMarker={this.addMarker} displayRoute={this.displayRoute}/>
        <div id="scroll" style={{height: 300}}>
          <ScrollMarkers ref={instance => {this.ScrollMarkers = instance;}} markers={this.state.markers}/>
        </div>
      </div>
    );
  }
}

export default Maps;