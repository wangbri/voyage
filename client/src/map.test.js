import React from 'react';
import ReactDOM from 'react-dom';
import Maps from './views/Maps';
import UserInput from './views/UserInput';
import renderer from 'react-test-renderer';
import ScrollMarkers from './components/Marker';
import Forms from './components/Forms';
import { locationsMarkers } from './api/googleMaps';


const setupGoogleMock = () => {
  /*** Mock Google Maps JavaScript API ***/
  const google = {
    maps: {
      Map:class{ zoom(){} center(){}},
      places: {
        AutocompleteService: () => {},
        PlacesServiceStatus: {
          INVALID_REQUEST: 'INVALID_REQUEST',
          NOT_FOUND: 'NOT_FOUND',
          OK: 'OK',
          OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
          REQUEST_DENIED: 'REQUEST_DENIED',
          UNKNOWN_ERROR: 'UNKNOWN_ERROR',
          ZERO_RESULTS: 'ZERO_RESULTS',
        },
      },
      Geocoder: () => {},
      GeocoderStatus: {
        ERROR: 'ERROR',
        INVALID_REQUEST: 'INVALID_REQUEST',
        OK: 'OK',
        OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
        REQUEST_DENIED: 'REQUEST_DENIED',
        UNKNOWN_ERROR: 'UNKNOWN_ERROR',
        ZERO_RESULTS: 'ZERO_RESULTS',
      },

      Geocoder:class{},
      DirectionsService:class{},
      DirectionsRenderer:class{ setMap(){}},
    },
  };

  global.window.google = google;
};


// in test file.
beforeAll(() => {
  setupGoogleMock();
});

//test if each component redered
it('Map renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Maps />, div);
});

it('User Input renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UserInput/>, div);
});

it('Forms renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Forms />, div);
});

it('Location Markers renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<locationsMarkers />, div);
});

// it('Scroll Markers renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<ScrollMarkers />, div);
// });

//test if each component matches its snapshot
it('User Input renders a snapshot', () => {
  const tree = renderer.create(<UserInput/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Map Input renders a snapshot', () => {
  const tree = renderer.create(<Maps />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Forms Input renders a snapshot', () => {
  const tree = renderer.create(<Forms />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Location Marker renders a snapshot', () => {
  const tree = renderer.create(<locationsMarkers />).toJSON();
  expect(tree).toMatchSnapshot();
});