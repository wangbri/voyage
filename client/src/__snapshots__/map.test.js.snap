// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms Input renders a snapshot 1`] = `
<div
  className="row"
  id="submitAddress"
>
  <input
    className="form-control"
    id="inputYelpAddress"
    onChange={[Function]}
    placeholder="Specify a Yelp address.."
    type="text"
    value=""
  />
  <div
    class="btn-group"
  >
    <button
      className="btn btn-primary"
      id="yelp-btn"
      onClick={[Function]}
      type="button"
    >
      Submit Yelp Address
    </button>
    <button
      className="btn btn-primary"
      id="generate-schedule-btn"
      onClick={[Function]}
      type="button"
    >
      Generate Schedule
    </button>
  </div>
</div>
`;

exports[`Location Marker renders a snapshot 1`] = `<locationsMarkers />`;

exports[`Map Input renders a snapshot 1`] = `
<div>
  <div
    id="map"
    style={
      Object {
        "height": 550,
      }
    }
  />
  <div
    className="row"
    id="submitAddress"
  >
    <input
      className="form-control"
      id="inputYelpAddress"
      onChange={[Function]}
      placeholder="Specify a Yelp address.."
      type="text"
      value=""
    />
    <div
      class="btn-group"
    >
      <button
        className="btn btn-primary"
        id="yelp-btn"
        onClick={[Function]}
        type="button"
      >
        Submit Yelp Address
      </button>
      <button
        className="btn btn-primary"
        id="generate-schedule-btn"
        onClick={[Function]}
        type="button"
      >
        Generate Schedule
      </button>
    </div>
  </div>
  <div
    id="scroll"
    style={
      Object {
        "height": 300,
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="scroll-horizontal "
        onWheel={[Function]}
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "display": "inline-flex",
              "height": "100%",
              "position": "absolute",
              "transform": "translate3d(0px, 0,0)",
              "willChange": "transform",
            }
          }
        >
          <div
            style={
              Object {
                "height": "50em",
                "margin": "10px",
                "width": "20em",
              }
            }
          >
            <div
              className="card"
              styles="width: 18rem;"
            >
              <img
                className="card-img-top"
                src="https://placeholdit.imgix.net/~text?txtsize=16&txt=318%C3%97180&w=288&h=50"
                styles="width: 18rem;"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                />
                <p
                  className="card-text"
                >
                  Some quick example text to build on the card title and make up the bulk of the card's content.
                </p>
                <a
                  className="btn btn-primary"
                  href="#"
                >
                  Go somewhere
                </a>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "height": "50em",
                "margin": "10px",
                "width": "20em",
              }
            }
          >
            <div
              className="card"
              styles="width: 18rem;"
            >
              <img
                className="card-img-top"
                src="https://placeholdit.imgix.net/~text?txtsize=16&txt=318%C3%97180&w=288&h=50"
                styles="width: 18rem;"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                />
                <p
                  className="card-text"
                >
                  Some quick example text to build on the card title and make up the bulk of the card's content.
                </p>
                <a
                  className="btn btn-primary"
                  href="#"
                >
                  Go somewhere
                </a>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "height": "50em",
                "margin": "10px",
                "width": "20em",
              }
            }
          >
            <div
              className="card"
              styles="width: 18rem;"
            >
              <img
                className="card-img-top"
                src="https://placeholdit.imgix.net/~text?txtsize=16&txt=318%C3%97180&w=288&h=50"
                styles="width: 18rem;"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                />
                <p
                  className="card-text"
                >
                  Some quick example text to build on the card title and make up the bulk of the card's content.
                </p>
                <a
                  className="btn btn-primary"
                  href="#"
                >
                  Go somewhere
                </a>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "height": "50em",
                "margin": "10px",
                "width": "20em",
              }
            }
          >
            <div
              className="card"
              styles="width: 18rem;"
            >
              <img
                className="card-img-top"
                src="https://placeholdit.imgix.net/~text?txtsize=16&txt=318%C3%97180&w=288&h=50"
                styles="width: 18rem;"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                />
                <p
                  className="card-text"
                >
                  Some quick example text to build on the card title and make up the bulk of the card's content.
                </p>
                <a
                  className="btn btn-primary"
                  href="#"
                >
                  Go somewhere
                </a>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "height": "50em",
                "margin": "10px",
                "width": "20em",
              }
            }
          >
            <div
              className="card"
              styles="width: 18rem;"
            >
              <img
                className="card-img-top"
                src="https://placeholdit.imgix.net/~text?txtsize=16&txt=318%C3%97180&w=288&h=50"
                styles="width: 18rem;"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                />
                <p
                  className="card-text"
                >
                  Some quick example text to build on the card title and make up the bulk of the card's content.
                </p>
                <a
                  className="btn btn-primary"
                  href="#"
                >
                  Go somewhere
                </a>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "height": "50em",
                "margin": "10px",
                "width": "20em",
              }
            }
          >
            <div
              className="card"
              styles="width: 18rem;"
            >
              <img
                className="card-img-top"
                src="https://placeholdit.imgix.net/~text?txtsize=16&txt=318%C3%97180&w=288&h=50"
                styles="width: 18rem;"
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                />
                <p
                  className="card-text"
                >
                  Some quick example text to build on the card title and make up the bulk of the card's content.
                </p>
                <a
                  className="btn btn-primary"
                  href="#"
                >
                  Go somewhere
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`User Input renders a snapshot 1`] = `<div />`;
